tests/integration/local/invoke/test_integrations_cli.py

```diff
@@ -376,8 +375,23 @@ def test_caching_two_layers(self):
 
         self.assertEquals(2, len(os.listdir(str(self.layer_cache))))
 
-    def test_layer_does_not_exist(self):
 
+class TestLayerVersionThatDoNotCreateCache(InvokeIntegBase):
+    template = Path("layers", "layer-template.yml")
+    region = 'us-west-2'
+    layer_utils = LayerUtils(region=region)
+
+    def setUp(self):
+        self.layer_cache = Path().home().joinpath("integ_layer_cache")
+
+    def tearDown(self):
+        docker_client = docker.from_env()
+        samcli_images = docker_client.images.list(name='samcli/lambda')
+        for image in samcli_images:
+            docker_client.images.remove(image.id)
+
+    def test_layer_does_not_exist(self):
```

Comment:
So this test, ensures a cache is created for a non existent layer? Its not very explicit what it does.

Reply:
No. This is just moving existing tests that do not create the cache directory. These tests do not download or build an image and therefore the `layer_cache_dir` is not created. The class these were previously under, removed the layer_cache after every tests. Instead of making the tests ignore the delete errors (OSError when the cache does not exist), I moved this to a new class. That way we ensure we always have a clean cache and reduce the risk of a failure to one test not impacting another.

Reply:
Ok got it, thanks for the explanation. Just trying to capture this information somewhere.